src/main/java/duke/main/Processor.java

```diff
@@ -0,0 +1,80 @@
+package main.java.duke.main;
+
+import main.java.duke.exception.DukeException;
+
+import main.java.duke.task.DeadLine;
+import main.java.duke.task.Event;
+import main.java.duke.task.Task;
+import main.java.duke.task.ToDo;
+import main.java.duke.task.TaskList;
+
+/**
+ * Handles the logic flow of Duke.
+ */
+public class Processor {
+
+    /**
+     * Processes the run of the program.
+     * @param taskList The user's task list.
+     * @param storage The storage storing the user's saved task list.
+     * @param ui UI that handles interaction with user.
+     * @throws DukeException Exception can be thrown due to multiple reasons, such as
+     *                       invalid command, empty task description, empty date or
+     *                       wrong date format.
+     */
+    public static void process(TaskList taskList, Storage storage, Ui ui) throws DukeException {
+        ui.greetings();
+        while(Parser.hasNextLine()) {
```

Comment:
Might want to avoid deep nesting. Possible to separate the unusual conditions and exit separately in an earlier block.